middle_end/flambda2/to_cmm/to_cmm_env.ml

```diff
@@ -395,3 +369,39 @@ let flush_delayed_lets ?(entering_loop = false) env =
   in
   let flush e = flush pures_to_flush env.stages e in
   flush, { env with stages = []; pures = pures_to_keep }
+
+(* Creation *)
+
+let create offsets functions_info ~return_continuation ~exn_continuation
+    ~my_region =
+  let t =
+    { return_continuation;
+      exn_continuation;
+      offsets;
+      functions_info;
+      stages = [];
+      pures = Variable.Map.empty;
+      vars = Variable.Map.empty;
+      vars_extra = Variable.Map.empty;
+      conts = Continuation.Map.empty;
+      exn_handlers = Continuation.Set.singleton exn_continuation;
+      exn_conts_extra_args = Continuation.Map.empty
+    }
+  in
+  (* Dummy binding for [my_region] *)
```

Comment:
I'm not sure I understand this binding: is it correct to put a dummy value here because: 1) the region is not used effectively used in local allocs, 2) the region is only really used when closing/ending a region, and it happens that we never close the function region ?

In any case, I'm also worried about why we state that the effects are pure for this expression.

And lastly, this kind of dummy binding would actually be a problem with #807 which tries to systematically inline bindings such as `let x = 0 in ...` since basically there's no real reason to put an integer in a variable (since it's better to use the integer constant at every use of the variable).

Reply:
If I understand correctly, the variable is never used (the only primitive taking region variables is `End_region`, and you're not supposed to close your caller's region). So I'd like to know why this binding was added, and if we can do without we should.

Reply:
I have submitted a patch (see https://github.com/mshinwell/flambda-backend/pull/17 ) to remove that unneeded binding. However, with my patch, we have failures because there are at least some unneeded re-bindings of `my_region` (e.g. a `let region = my_region in <body>` where `region` is actually unused in body), that although useless, are still translated to cmm, and cause errors later in the backend.